google/cloud/opentelemetry/internal/recordable.cc

```diff
@@ -52,6 +53,23 @@ void MapKey(opentelemetry::nostd::string_view& key) {
   if (it != m->end()) key = it->second;
 }
 
+template <typename T>
+std::string ToString(opentelemetry::nostd::span<T const> values) {
+  return absl::StrJoin(values, "|");
```

Comment:
Whether `|` is a good separator depends on the context.  Maybe formatting as an array is better:

```cc
return absl::StrCat("[", absl::StrJoin(values, ", "), "]")
```

In some contexts `\003` (ETX) may be appropriate:

https://en.wikipedia.org/wiki/End-of-Text_character

or one of the "separators" (`\x1C' to `\x1F`):

https://en.wikipedia.org/wiki/C0_and_C1_control_codes

Though that would make the string non-printable.

Reply:
I don't think there is a perfect answer. We will go with your array suggestion.